src/client/src/app/site/[site]/availability/edit-services/confirmation/page.tsx

```diff
@@ -0,0 +1,111 @@
+import {
+  assertPermission,
+  fetchSite,
+  availabilityChangeProposal,
+  fetchClinicalServices,
+} from '@services/appointmentsService';
+import {
+  AvailabilityChangeProposalRequest,
+  SessionSummary,
+  AvailabilitySession,
+} from '@types';
+import { parseToUkDatetime, toTimeFormat } from '@services/timeService';
+import { notFound } from 'next/navigation';
+import NhsTransactionalPage from '@components/nhs-transactional-page';
+import fromServer from '@server/fromServer';
+import { EditServicesConfirmationPage } from './edit-services-confirmation';
+
+type PageProps = {
+  searchParams?: Promise<{
+    date: string;
+    removedServicesSession: string;
+    session: string;
+    sessionToEdit: string;
+  }>;
+  params: Promise<{
+    site: string;
+  }>;
+};
+
+const Page = async ({ searchParams, params }: PageProps) => {
+  const { site: siteFromPath } = { ...(await params) };
+  const { removedServicesSession, date, sessionToEdit, session } = {
+    ...(await searchParams),
+  };
+
+  if (
+    removedServicesSession === undefined ||
+    date === undefined ||
+    sessionToEdit === undefined ||
+    session === undefined
+  ) {
+    return notFound();
+  }
+
+  await assertPermission(siteFromPath, 'availability:setup');
+  const parsedDate = parseToUkDatetime(date);
+  const site = await fromServer(fetchSite(siteFromPath));
+  const sessionSummary: SessionSummary = JSON.parse(atob(session));
+  const removedServicesSessionDetails: AvailabilitySession = JSON.parse(
+    atob(removedServicesSession),
+  );
+
+  const newSessionDetails: AvailabilitySession = JSON.parse(
+    atob(decodeURIComponent(sessionToEdit)),
+  );
+
+  newSessionDetails.services = newSessionDetails.services.filter(
+    service => !removedServicesSessionDetails.services.includes(service),
+  );
+
+  const availabilityRequest: AvailabilityChangeProposalRequest = {
+    from: date,
+    to: date,
+    site: siteFromPath,
+    sessionMatcher: {
+      from: toTimeFormat(sessionSummary.ukStartDatetime) ?? '',
+      until: toTimeFormat(sessionSummary.ukEndDatetime) ?? '',
+      services: Object.keys(sessionSummary.totalSupportedAppointmentsByService),
+      slotLength: sessionSummary.slotLength,
+      capacity: sessionSummary.capacity,
+    },
+    sessionReplacement: {
+      from: removedServicesSessionDetails.from,
+      until: removedServicesSessionDetails.until,
+      services: newSessionDetails.services,
+      slotLength: removedServicesSessionDetails.slotLength,
+      capacity: removedServicesSessionDetails.capacity,
+    },
+  };
+
+  const availabilityProposal = await fromServer(
+    availabilityChangeProposal(availabilityRequest),
+  );
+
+  const clinicalServices = await fromServer(fetchClinicalServices());
+
```

Comment:
This fetch could be fairly slow, I think we should wrap the `<EditServicesConfirmationPage/>` component in a Suspense with spinner so the user sees a spinner if this takes a second or two to load. There's a working example of this pattern in `view-availability-page.tsx` amongst others

Reply:
(When first introducing this pattern I found the easiest way to test it was just to lob a Thread.Sleep(5000) in the backend API it calls, then walk the route manually)